ui/src/app/cohort-search/selection-list/selection-list.component.tsx

```diff
@@ -0,0 +1,232 @@
+}
+
+interface SelectionInfoState {
+  truncated: boolean;
+}
+
+export class SelectionInfo extends React.Component<SelectionInfoProps, SelectionInfoState> {
+  name: HTMLDivElement;
+  constructor(props: SelectionInfoProps) {
+    super(props);
+    this.state = {truncated: false};
+  }
+
+  componentDidMount(): void {
+    const {offsetWidth, scrollWidth} = this.name;
+    this.setState({truncated: scrollWidth > offsetWidth});
+  }
+
+  get showType() {
+    return ![
+      DomainType.PHYSICALMEASUREMENT.toString(),
+      DomainType.DRUG.toString(),
+      DomainType.SURVEY.toString()
+    ].includes(this.props.selection.domainId);
+  }
+
+  get showOr() {
+    const {index, selection} = this.props;
+    return index > 0 && selection.domainId !== DomainType.PERSON.toString();
+  }
+
+  render() {
+    const {selection, removeSelection} = this.props;
+    const itemName = <React.Fragment>
+      {this.showType && <strong>{typeDisplay(selection)}&nbsp;</strong>}
+      {nameDisplay(selection)} {attributeDisplay(selection)}
+    </React.Fragment>;
+    return <div style={styles.selectionItem}>
+      <button style={styles.removeSelection} onClick={() => removeSelection()}>
+        <ClrIcon shape='times-circle'/>
+      </button>
+      <div style={styles.itemInfo}>
+        {this.showOr && <strong>OR&nbsp;</strong>}
+        {!!selection.group && <span>Group&nbsp;</span>}
+        <TooltipTrigger disabled={!this.state.truncated} content={itemName}>
+          <div style={styles.itemName} ref={(e) => this.name = e}>
+            {itemName}
+          </div>
+        </TooltipTrigger>
+      </div>
+    </div>;
+  }
+}
+
+interface Props {
+  back: Function;
+  cancel: Function;
+  disableFinish: boolean;
+  domain: DomainType;
+  finish: Function;
+  removeSelection: Function;
+  selections: Array<Selection>;
+  setView: Function;
+  view: string;
+}
+
+export class SelectionList extends React.Component<Props> {
+  constructor(props: Props) {
+    super(props);
+  }
+
+  get showModifiers() {
+    return ![DomainType.PHYSICALMEASUREMENT, DomainType.PERSON, DomainType.SURVEY].includes(this.props.domain);
+  }
+
+  get showNext() {
+    return this.showModifiers && this.props.view !== 'modifiers';
+  }
+
+  get showBack() {
+    return this.showModifiers && this.props.view === 'modifiers';
+  }
+
+  render() {
+    const {back, cancel, disableFinish, finish, removeSelection, selections, setView} = this.props;
+    return <div style={styles.selectionPanel}>
+      <h5 style={styles.selectionTitle}>Selected Criteria</h5>
+      <div style={styles.selectionContainer}>
+        {selections.map((selection, s) =>
+          <SelectionInfo key={s}
+            index={s}
+            selection={selection}
+            removeSelection={() => removeSelection(selection)}/>
+        )}
+      </div>
+      <div style={styles.buttonContainer}>
+        <button style={{...styles.button, background: 'transparent', color: colors.dark, fontSize: '14px'}}
```

Comment:
Consider using the `Button` component in `buttons.tsx`

Reply:
Good idea, switch all 4 buttons here to use `Button`